### PR TITLE
feat: github repository IaC

### PR DESCRIPTION
**Pull request by zackpollard**
With this, we now control creation and management of repositories with IaC. This includes the creation of the repo, creation of our default rulesets for main protection and creation of some default files we should have everywhere

### CODE REVIEW

**Review by jrasm91 — APPROVED** [review]

File: deployment/modules/github/org/repositories.tf [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+resource "github_repository" "repositories" {
+  for_each                  = { for repo in var.repositories : repo.name => repo }
+  name                      = each.value.name
+  description               = each.value.description
+  allow_auto_merge          = true
+  allow_merge_commit        = false
+  allow_rebase_merge        = false
+  allow_squash_merge        = true
+  allow_update_branch       = true
+  archived                  = false
```
Comment (bo0tzz) [inline deployment/modules/github/org/repositories.tf:41]: This'll override cli from the list above

```
  # github_repository.repositories["CLI"] will be updated in-place
  # (imported from "CLI")
  ~ resource "github_repository" "repositories" {
      (...)
      ~ archived                    = true -> false
```

**Review by bo0tzz — COMMENTED** [review]
The plan lists an extensive diff for the rulesets on the main `immich` repo. I think that's because we're currently using branch protection rules rather than rulesets. Are you handling that switch manually?

**Comment by zackpollard** [comment]
> The plan lists an extensive diff for the rulesets on the main `immich` repo. I think that's because we're currently using branch protection rules rather than rulesets. Are you handling that switch manually?

Yea, I will be removing the existing branch protections manually when this is merged, then moving the github status checks to a ruleset we will manually maintain for the time being

File: deployment/modules/github/org/repositories.tf (hunk 1 above)
Comment (zackpollard) [reply deployment/modules/github/org/repositories.tf]: Fixed

**Pull request merged**